src/js/components/BatchContainer.js

```diff
@@ -0,0 +1,63 @@
+import React, { Component } from "react";
+
+import Transaction from "#SRC/js/structs/Transaction";
+import TransactionTypes from "#SRC/js/constants/TransactionTypes";
+
+export default class Batch extends Component {
+  constructor() {
+    super(...arguments);
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleAddItem = this.handleAddItem.bind(this);
+    this.handleRemoveItem = this.handleRemoveItem.bind(this);
+  }
+
+  handleChange(event) {
+    const fieldName = event.target.getAttribute("name");
+    if (!fieldName) {
+      return;
+    }
+
+    const value = event.target.type === "checkbox"
+      ? event.target.checked
+      : event.target.value;
+
+    const path = fieldName.split(".");
+    const batch = this.props.batch.add(new Transaction(path, value));
+
+    this.props.onChange(batch, fieldName);
+  }
+
+  handleAddItem(event) {
+    const { value, path } = event;
+
+    const batch = this.props.batch.add(
+      new Transaction(path.split("."), value, TransactionTypes.ADD_ITEM)
+    );
+
+    this.props.onChange(batch);
+  }
+
+  handleRemoveItem(event) {
+    const { value, path } = event;
+
+    const batch = this.props.batch.add(
+      new Transaction(path.split("."), value, TransactionTypes.REMOVE_ITEM)
+    );
+
+    this.props.onChange(batch);
+  }
+
+  render() {
+    const { children } = this.props;
+
+    return (
+      <div onChange={this.handleChange}>
+        {React.cloneElement(children, {
```

Comment:
💭   Not blocking but we should check if there are other (better) ways than cloning to _connect_ two elements as we the _surrounding_  component might overwrite props directly passed to the children. /cc @dcos/frontend-foundation

Reply:
We can use react context - it's like props, but if you need to go deeper -_-
But this is the technique we're using all over the place in the router. We just clone elements and mix props we want to pass down.

Reply:
I know that we're using this technique in many places and I think it's about time that we question the approach and think about a different solution.

Reply:
Even though pitfalls are pretty obvious, I'd say there's not much you can do. Overusing context is also considered a bad practice.